ui/src/app/applications/components/application-details/application-resource-list.tsx

```diff
@@ -1,104 +1,147 @@
-                <div className='columns small-2 xxxlarge-2'>NAME</div>
-                <div className='columns small-2 xxxlarge-2'>GROUP/KIND</div>
-                <div className='columns small-1 xxxlarge-2'>SYNC ORDER</div>
-                <div className='columns small-2 xxxlarge-2'>NAMESPACE</div>
-                <div className='columns small-2 xxxlarge-2'>CREATED AT</div>
-                <div className='columns small-2 xxxlarge-2'>STATUS</div>
+    tree?: models.ApplicationTree;
+}) => {
+    function getResNode(nodes: ResourceNode[], nodeId: string): models.ResourceNode {
+        for (const node of nodes) {
+            if (nodeKey(node) === nodeId) {
+                return node;
+            }
+        }
+        return null;
+    }
+    const parentNode = ((resources || []).length > 0 && (getResNode(tree.nodes, nodeKey(resources[0])) as ResourceNode)?.parentRefs?.[0]) || ({} as ResourceRef);
+
+    return (
+        <div className='argo-table-list argo-table-list--clickable'>
+            <div className='argo-table-list__head'>
+                <div className='resource-details__header' style={{paddingTop: '20px'}}>
+                    {Object.keys(parentNode).length > 0 && (
+                        <React.Fragment>
+                            <div className='resource-parent-node-info'>
+                                <label className='resource-parent-node-label'>Parent Node Info</label>
+                                <label className='resource-parent-node-label'>Name: {parentNode?.name}</label>
+                                <label className='resource-parent-node-label'>Kind: {parentNode?.kind}</label>
+                            </div>
+                        </React.Fragment>
+                    )}
+                </div>
+                <div className='row'>
+                    <div className='columns small-1 xxxlarge-1' />
+                    <div className='columns small-2 xxxlarge-2'>NAME</div>
+                    <div className='columns small-1 xxxlarge-1'>GROUP/KIND</div>
+                    <div className='columns small-1 xxxlarge-1'>SYNC ORDER</div>
+                    <div className='columns small-2 xxxlarge-2'>NAMESPACE</div>
+                    {(parentNode.kind === 'Rollout' || parentNode.kind === 'Deployment') && <div className='columns small-1 xxxlarge-1'>REVISION</div>}
+                    <div className='columns small-2 xxxlarge-2'>CREATED AT</div>
+                    <div className='columns small-1 xxxlarge-1'>STATUS</div>
+                </div>
             </div>
-        </div>
-        {resources
-            .sort((first, second) => -createdOrNodeKey(first).localeCompare(createdOrNodeKey(second)))
-            .map(res => (
-                <div
-                    key={nodeKey(res)}
-                    className={classNames('argo-table-list__row', {
-                        'application-resource-tree__node--orphaned': res.orphaned
```

Comment:
How about the orphaned resources?

Reply:
My change is specific to listing the resources which are grouped. The change should not change the view of orphan resources. The list view at the application detail should continue displaying the orphan resource.